File: ios/Sources/TectonicCapacitorUtilsPlugin/TectonicCapacitorUtilsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(TectonicCapacitorUtilsPlugin)
public class TectonicCapacitorUtilsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "TectonicCapacitorUtilsPlugin"
    public let jsName = "TectonicCapacitorUtils"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getAPNSToken", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = TectonicCapacitorUtils.shared

    @objc func getAPNSToken(_ call: CAPPluginCall) {
        let token = implementation.getAPNSToken()
        call.resolve([
            "token": token ?? NSNull()
        ])
    }
}

File: android/src/main/java/com/tectonic/capacitor/utils/TectonicCapacitorUtilsPlugin.java
package com.tectonic.capacitor.utils;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "TectonicCapacitorUtils")
public class TectonicCapacitorUtilsPlugin extends Plugin {

    private TectonicCapacitorUtils implementation = new TectonicCapacitorUtils();

    @PluginMethod
    public void getAPNSToken(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("token", implementation.getAPNSToken());
        call.resolve(ret);
    }
}
